test_suite/package/test_otel.py

```diff
@@ -0,0 +1,132 @@
+import pytest
+import logging
+import time
+import json
+
+from test_suite.generic.test_generic import TestsSubscriptionManager
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def check_instance_status(request, instance_data, host):
+    instance_id = instance_data['instance_id']
+    command_to_run = [
+        'aws', 'ec2', 'describe-instance-status',
+        '--instance-ids', instance_id
+    ]
+
+    while True:
+        command_output = host.backend.run_local(" ".join(command_to_run)).stdout
+        cmd_output = json.loads(command_output)
+
+        instance_status = cmd_output["InstanceStatuses"][0]["InstanceStatus"]["Status"]
+        system_status = cmd_output["InstanceStatuses"][0]["SystemStatus"]["Status"]
+
+        if instance_status == "ok" and system_status == "ok":
+            break
+        logger.info("Instance status is not 'passed' yet. Waiting...")
+        time.sleep(5)
+
+
+@pytest.fixture()
+def modify_iam_role(request, instance_data, host):
+    instance_id = instance_data['instance_id']
+    iam_role_name = "CloudWatchAgentServerRole_2"
+
+    command_to_run = [
+        'aws', 'ec2', 'associate-iam-instance-profile',
+        '--instance-id', instance_id,
+        '--region', 'us-west-1',
+        '--iam-instance-profile', 'Name="{}"'.format(iam_role_name)
+    ]
+    modify_iam_role_cmd = ' '.join(command_to_run)
+
+    assert host.backend.run_local(modify_iam_role_cmd), 'faild to update iam role'
+
+
+@pytest.fixture()
+def install_packages(request, host):
+    class_instance = request.node.cls
+    install_cmd = f'yum install -y  {class_instance.package_name} --nogpgcheck'
+    repo_path = ('https://copr.fedorainfracloud.org/coprs/miyunari'
+                 '/redhat-opentelemetry-collector/repo/rhel-9/miyunari-redhat-opentelemetry-collector-rhel-9.repo')
+    with host.sudo():
+        assert host.run_test(f'yum-config-manager --add-repo {repo_path}')
+        assert host.run_test(install_cmd), f'Failed to install the package {class_instance.package_name}'
+        output = host.check_output("rpm -qa | grep opentelemetry*")
```

Comment:
Please use `pytest-testinfra` modules whenever possible, to avoid having to manually run commands for things that our framework already do for you.
Please check [host.package](https://testinfra.readthedocs.io/en/latest/modules.html#testinfra.modules.package.Package)

Reply:
Maybe in this case it is because you want to list all of the matching packages, in that case I think it's perfect. I would just suggest to use `test_lib.print_host_command_output()`. It will automatically print all the packages for your so you don't have to store cmd output in a var to later print it out.

```suggestion
        test_lib.print_host_command_output("rpm -qa | grep opentelemetry*")
```

Reply:
regarding the other I couldn't find in test infra commands that support um install or config manager, so I will leave it as it is (line 54,55)
If you find something, so please update me!:)

Reply:
Yeah, I think there's nothing related to adding repos and installing packages, only querying them. In this case it makes total sense to query all packages to then get a list with grep, that maches all the OTEL-related pkgs.

The package module can query only one specific package with a specific name, so that would mean we need to create a list of otel pkg names to check and iterate over them. Not a fan of that.